src/routes/users.rs

```diff
@@ -41,16 +41,39 @@ pub async fn users_get(
     web::Query(ids): web::Query<UserIds>,
     pool: web::Data<PgPool>,
 ) -> Result<HttpResponse, ApiError> {
-    let user_ids = serde_json::from_str::<Vec<UserId>>(&ids.ids)?
-        .into_iter()
-        .map(|x| x.into())
-        .collect();
-
-    let users_data = User::get_many(user_ids, &**pool).await?;
+    let mut users_data;
+
+    let decoded_user_ids = serde_json::from_str::<Vec<UserId>>(&ids.ids);
```

Comment:
due to usernames often being longer than 10 characters, this will throw an error.
And we therefore search on usernames instead, not sure if (also see media) that makes the most sense? It would of course be explained in the API Docs, could be I'm missing some fancy way of doing this, that could not lead to the same slightly "unexpected" behaviour.

Reply:
That seems like a really poor, bug-prone way of doing it. Isn't there a get_project_from_id_or_slug method (or something similar) for projects? How is that implemented?

Reply:
There is not.
This is similar to how fetching a single user is implemented. 
```rust
    let string = info.into_inner().0;
    let id_option: Option<UserId> =
        serde_json::from_str(&format!("\"{string}\"")).ok();

    let mut user_data;

    if let Some(id) = id_option {
        user_data = User::get(id.into(), &**pool).await?;

        if user_data.is_none() {
            user_data = User::get_from_username(string, &**pool).await?;
        }
    } else {
        user_data = User::get_from_username(string, &**pool).await?;
    }

    if let Some(data) = user_data {
        let response: crate::models::users::User = data.into();
        Ok(HttpResponse::Ok().json(response))
    } else {
        Ok(HttpResponse::NotFound().body(""))
    }
```

Reply:
That isn't what the code actually does. It checks if it is parable as an ID (as ids always take precedence over a slug). If a project isn't found by ID, it tries to find it using a slug.

Reply:
It it though.

The single user get
Tries to first parse the string into a UserId. (This would throw an error if the value is greater than 10 characters), since we directly calling `.ok()` the error is discarded, but a None value is then provided.

If it matched an id (less than 11), meaning the string contains a value (which technically could be a small username), it tries to first find a user from id, if that fails (due to low character username), then it fetches on username.

If originally the value was more than 10 characters, (often a username) it would go straight to finding a user based on their username.

Which is identical to what I'm doing here for multiple users.
I first try and parse the list of values into a list of user_ids, if that fails (errors, since I'm not calling `.ok()` directly) then I go straight to fetching users on their usernames.

if it doesn't fail originally, it would automatically go into and try and fetch users based on their ids. (At this point that array could be a mix of id's and low character usernames)
If the list contains ANY id's, the get_many call will find at least one, and exit.
If the list ONLY contains low character names, then the get_many (based on id's) will not fetch any, trying to fetch users based on usernames.

I'm going to say that logic is identical, between the single user and multiple users. (Except for the case where the array is mixed).

_On a completely side note, I would say this logic should be abstracted away to a method inside the User model, since the "controller/router" does not need to care about this. it should just know if any users is retrieved and then return them._